### ISSUE TITLE
CreateFFmpegInteropMSS methods will get out of hand if we add a lot more options

### ISSUE THREAD

**Issue by lukasf** [Microsoft/FFmpegInterop#174]
There are already a whole bunch of parameters already in these methods. I have some new features in mind which would require to add even more parameters. If we continue to evolve the library and add all new options here, we will run into problems, this will be confusing.

I would propose to move to a different initialization model, like this:

```
var ffmpeg = new FFmpegInteropMSS();
ffmpeg.ForceVideoDecode = false;
ffmpeg.ForceAudioDecode = true;
ffmpeg.AllowMultipleAudioStreams = true;
ffmpeg.CorruptPackageThreshold = 100;
await ffmpeg.OpenStreamAsync(stream);  // this could be async to prevent UI blocking
var mss = ffmpeg.GetMediaStreamSource();
```

This is more like what you expect from framework WinRT classes.
The old CreateFFmpegInteropMSS could be left for compatibility reasons.

What do you think?

**Comment by khouzam** [comment]
That sounds very reasonable. I'd love to start the discussion about this. I do want this project to move in the direction that the community wants.

**Comment by mcosmin222** [comment]
Maybe we should have a builder instead of exposing them directly to the MSS

**Comment by lukasf** [comment]
I see FFmpegInteropMSS as the builder, and you get the result with GetMediaStreamSource() (which is what you are really interested in).

**Comment by lukasf** [comment]
Actually, I'd rename GetMediaStreamSource() to CreateMediaStreamSource(), so after OpenAsync() you could inspect and configure the streams (in case you have multiple audio streams, you could e.g. set force decode per-stream), and then finally CreateMediaStreamSource().

**Comment by mcosmin222** [comment]
Not really, the FFMpegInteropMSS also has some stuff needed by the MSS to work properly (like handling its events). So it is sort of like a builder but more than a builder.

**Comment by lukasf** [comment]
True, it is both builder and stream handler. But opening a file and handling MSS events is very closely coupled, I am not sure if this can be separated in a good way. Or if you see the builder as just collecting property values and passing them to FFMpegInteropMSS constructor then I do not see so much improvement in that.

**Comment by mcosmin222** [comment]
The problem with having public setters for properties is that the ffmpeginterop object is immutable in nature. You can't just set ForceAudioDecode mid way during playback. Doing that would force you to recreate the entire media stream source, and in order for it to be transparent for consumers, you would also have to seek to its last known position. 

Sure, some properties can have setters, but the heavy hitters which are currently passed in constructors, can't.

**Comment by lukasf** [comment]
You have a point there. We could throw a meaningful exception when setting a property after call to CreateMediaStreamSource() but I agree that it is not very intuitive.

**Comment by mcosmin222** [comment]
Maybe have a settings class you can pass to the constructor and have the ffmpeginterop object have private setters.
